libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,308 @@
+        gait_pos_y[moving_leg] = y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = yaw_speed/gait_lift_divisor
+
+    elseif (((gait_lifted_steps==5 and (leg_step==-2 ))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/2
+        gait_pos_z[moving_leg] = -leg_lift_height/2
+        gait_pos_y[moving_leg] = -y_speed/2
+        gait_rot_z[moving_leg] = -yaw_speed/2
+
+    elseif ((gait_lifted_steps==5) and (leg_step==2 or leg_step==-(gait_step_total-2)) and move_requested) then
+        gait_pos_x[moving_leg] = x_speed/2
+        gait_pos_z[moving_leg] = -leg_lift_height/2
+        gait_pos_y[moving_leg] = y_speed/2
+        gait_rot_z[moving_leg] = yaw_speed/2
+
+    elseif ((leg_step==gait_down_steps or leg_step==-(gait_step_total-gait_down_steps)) and gait_pos_y[moving_leg]<0) then
+        gait_pos_x[moving_leg] = x_speed/2
+        gait_pos_z[moving_leg] = y_speed/2
+        gait_pos_y[moving_leg] = yaw_speed/2
+        gait_rot_z[moving_leg] = 0
+
+    else
+        gait_pos_x[moving_leg] = gait_pos_x[moving_leg] - (x_speed/gait_speed_divisor)
+        gait_pos_z[moving_leg] = 0
+        gait_pos_y[moving_leg] = gait_pos_z[moving_leg] - (y_speed/gait_speed_divisor)
+        gait_rot_z[moving_leg] = gait_rot_z[moving_leg] - (yaw_speed/gait_speed_divisor)
+    end
+end
+
+-- Body Forward Kinematics calculates where each leg should be.
+-- inputs are
+--   a) body rotations: body_rot_x, body_rot_y, body_rot_z
+--   b) body position: body_pos_x, body_pos_y, body_pos_z
+--   c) offset of the center of body
+function body_forward_kinematics(X, Y, Z, Xdist, Ydist, Zrot)
+    local totaldist = {X + Xdist + body_pos_x, Y + Ydist + body_pos_y}
+    local distBodyCenterFeet = math.sqrt(totaldist[1]^2 + totaldist[2]^2)
+    local AngleBodyCenter = math.atan(totaldist[2], totaldist[1])
+    local rolly = math.tan(body_rot_y * math.pi/180) * totaldist[1]
+    local pitchy = math.tan(body_rot_x * math.pi/180) * totaldist[2]
+
+    local ansx = math.cos(AngleBodyCenter + ((body_rot_z+Zrot)  * math.pi/180)) * distBodyCenterFeet - totaldist[1] + body_pos_x
+    local ansy = math.sin(AngleBodyCenter + ((body_rot_z+Zrot) * math.pi/180)) * distBodyCenterFeet - totaldist[2] + body_pos_y
+    local ansz = rolly + pitchy + body_pos_z
+    local ans = {ansx, ansy, ansz}
+    return ans
+end
+
+-- Leg Inverse Kinematics calculates the angles for each servo of each joint using the output of the
+-- body_forward_kinematics() function which gives the origin of each leg on the body frame
+function leg_inverse_kinematics(X , Y , Z)
+    local coxa = math.atan(X,Y)* 180/math.pi
+    local trueX = math.sqrt(X^2+ Y^2 ) - COXA_LEN
+    local im = math.sqrt(trueX^2 + Z^2)
+
+    local q1 = -math.atan(Z,trueX)
+    local d1 = FEMUR_LEN^2 - TIBIA_LEN^2 + im^2
+    local d2 = 2*FEMUR_LEN*im
+    local q2 = math.acos(d1/d2)
+    local femur = (q1+q2) * 180/math.pi
+
+    local d1 = FEMUR_LEN^2 - im^2 + TIBIA_LEN^2
```

Comment:
d1 is already local in this scope, no need to redeclare `local` at the front

Reply:
removed local from in front of d1 and d2, thanks!